src/libsyntax/diagnostics/macros.rs

```diff
@@ -1,13 +1,14 @@
 #[macro_export]
-macro_rules! register_diagnostic {
-    ($code:tt, $description:tt) => (__register_diagnostic! { $code, $description });
-    ($code:tt) => (__register_diagnostic! { $code })
+macro_rules! diagnostic_used {
+    ($code:ident) => (
+        let _ = crate::error_codes::$code;
```

Comment:
I might be wrong, but marking `mod error_codes` as `pub` would prevent this macro from being useful? We don't issue unused warnings for exported out of the crate things.